tests/testthat/test_jrepl.R

```diff
@@ -1,5 +1,22 @@
 test_that("works as expected", {
 
+    # duplication
+      
+      x = data.frame( 
+        id = as.integer( c( 1, 2, 3 ) ),
+        val = c( 1, 2, 3 ),
+        stringsAsFactors = TRUE
+      )
+      
+      y = data.frame(
+        id = as.integer( c( 2, 1, 1 ) ),
+        val = c( 1, 2, 3 ),
+        stringsAsFactors = TRUE
+      )
+      
+      expect_error( { t = jrepl( x, y, by = 'id', replace.cols = 'val' ) }, regexp = 'duplicated' )
+
+  # other tests:
```

Comment:
please also add a test that verifies the new return.all.dups option works as expected (returns a data from of dups matching the rows in x that would be duplicated).